uaclient/entitlements/repo.py

```diff
@@ -183,9 +183,10 @@ def purge_kernel_check(self, package_list):
         prompt the user for confirmation before proceeding.
         """
         linux_image_versions = [
-            package.name[len("linux-image-") :]
+            m.group(1)
             for package in package_list
-            if re.match(RE_KERNEL_PKG, package.name)
+            for m in [re.search(RE_KERNEL_PKG, package.name)]
+            if m
         ]
```

Comment:
I'm not a fan of multiple loops in a list comprehension, and in this case it's kind of a hack to iterate over a list of size one just to get the match variable and check if it's None.

I recommend something like:
```python
        linux_image_versions = []
        for package in package_list:
            m = re.search(RE_KERNEL_PKG, package.name)
            if m:
                linux_image_versions.append(m.group(1))
```

Reply:
agreed readability++